src/lib/hooks/useWorldState.svelte.ts

```diff
@@ -1,43 +1,76 @@
-import { toPath, getInUnsafe, mutInUnsafe } from '@thi.ng/paths'
 import {
 	WorldStateStoreClient,
-	type TransformChangeEvent,
 	TransformChangeType,
+	type TransformWithUUID,
+	ResourceName,
 } from '@viamrobotics/sdk'
 import {
 	createResourceClient,
 	createResourceQuery,
 	createResourceStream,
 	useResourceNames,
 } from '@viamrobotics/svelte-sdk'
-import { fromTransform, WorldObject } from '$lib/WorldObject.svelte'
+import { fromTransform } from '$lib/WorldObject.svelte'
 import { usePartID } from './usePartID.svelte'
+import { setInUnsafe } from '@thi.ng/paths'
+import type { ProcessMessage } from '$lib/world-state-messages'
+import { getContext, setContext } from 'svelte'
 
-export const useWorldStates = () => {
+const key = Symbol('world-state-context')
+
+interface Context {
+	names: ResourceName[]
+	current: Record<string, ReturnType<typeof createWorldState>>
+}
+
+const worker = new Worker(new URL('../workers/worldStateWorker.ts', import.meta.url), {
```

Comment:
Should we use vite's convienience stuff for web workers? I know that exporting libraries with them can get hairy: https://v3.vitejs.dev/guide/features.html#import-with-query-suffixes

Reply:
See the [above section](https://v3.vitejs.dev/guide/features.html#import-with-constructors): 

> A web worker script can be imported using [new Worker()](https://developer.mozilla.org/en-US/docs/Web/API/Worker/Worker) and [new SharedWorker()](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker/SharedWorker). Compared to the worker suffixes, this syntax leans closer to the standards and is the **recommended** way to create workers.